src/BootstrapTable.js

```diff
@@ -142,7 +142,13 @@ class BootstrapTable extends Component {
     const { options, selectRow } = nextProps;
 
     this.store.setData(nextProps.data.slice());
-    let page = options.page || this.state.currPage;
+
+    let page;
+    if (options.page !== 'undefined') {
```

Comment:
You should use `typeof options.page !== 'undefined'`. Because the statement you use, does not do what you expect! the value `undefined` is never `===` the string `'undefined'`!